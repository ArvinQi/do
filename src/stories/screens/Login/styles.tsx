import { Dimensions, StyleSheet } from "react-native";

const deviceHeight = Dimensions.get("window").height;

const styles: any = StyleSheet.create({
	container: {
		position: "absolute",
		top: 0,
		bottom: 0,
		left: 0,
		right: 0,
		backgroundColor: "#FBFAFA",
	},
	shadow: {
		flex: 1,
		width: undefined,
		height: undefined,
	},
	bg: {
		flex: 1,
		marginTop: deviceHeight / 1.75,
		paddingTop: 20,
		paddingLeft: 10,
		paddingRight: 10,
		paddingBottom: 30,
		bottom: 0,
	},
	logo: {
		width: 200,
		height: 200,
		marginTop: 16,
	},
	input: {
		marginBottom: 20,
	},
	btn: {
		marginTop: 20,
		backgroundColor: "rgba(0,0,0, .6)",
	},
	content: {
		backgroundColor: "#FFF",
		marginTop: 0,
	},
});
export default styles;
