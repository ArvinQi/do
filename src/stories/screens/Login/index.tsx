import * as React from "react";
import { View, Image } from "react-native";
import { Container, Content, Footer, Button, Text } from "native-base";
import styles from "./styles";
export interface Props {
	loginForm: any;
	onLogin: Function;
	goRegister: Function;
}
export interface State { }
class Login extends React.Component<Props, State> {
	render() {
		return (
			<Container>
				<Content style={styles.content} padder>
					<View style={{ alignItems: "center" }}>
						<Image style={styles.logo} source={require("../../../../assets/do.png")}></Image>
					</View>
					{this.props.loginForm}
					<View>
						<Button style={styles.btn} block onPress={() => this.props.onLogin()}>
							<Text>Login</Text>
						</Button>
					</View>
				</Content>
				<Footer style={{ backgroundColor: "#F8F8F8" }}>
					<View style={{ alignItems: "center", opacity: 0.5, flexDirection: "row" }}>
						{/* <View padder> */}
							<Text style={{ color: "#ec4c4c" }}>Have none account? </Text>
							<Text onPress={() => this.props.goRegister()}>register now</Text>
						{/* </View> */}
					</View>
				</Footer>
			</Container>
		);
	}
}

export default Login;
