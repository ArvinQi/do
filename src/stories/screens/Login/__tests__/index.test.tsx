import React from "react";
import Login from "../index";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

const onLogin = jest.fn();
const goRegister = jest.fn();
const loginForm = React.Component;

it("renders correctly", () => {
	const tree = renderer.create(<Login onLogin={onLogin} loginForm={loginForm} goRegister={goRegister} />).toJSON();
	expect(tree).toMatchSnapshot();
});
