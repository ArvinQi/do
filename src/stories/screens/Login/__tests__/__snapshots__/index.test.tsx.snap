// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    onScroll={[Function]}
    padder={true}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
        Object {
          "backgroundColor": "#FFF",
          "marginTop": 0,
        },
      ]
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 200,
              "marginTop": 16,
              "width": 200,
            }
          }
        />
      </View>
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "rgba(0,0,0, .6)",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginTop": 20,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "lineHeight": 19,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#cbcbcb",
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "center",
          "left": 0,
          "right": 0,
        },
        Object {
          "backgroundColor": "#F8F8F8",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 0.5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#ec4c4c",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        Have none account? 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        register now
      </Text>
    </View>
  </View>
</View>
`;
